index.js

```diff
@@ -1,88 +1,125 @@
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+console.log("%cQuestion 3", "color: red") 
 
-// console.log(drinkWater(12))
+function drinkWater(currentThirstLevel) { 
+  console.log("Man I sure am thirsty")
+  currentThirstLevel -= 1
+  console.log( 'Ahh that hits the spot' )
+  return currentThirstLevel
+}
+ console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red") 
+
+function sameSameButDifferent(num, maybeNum) {
+  if (num === maybeNum) {
+    console.log('same same')
+  } else if (num === parseInt(maybeNum, 10)) {
+    console.log( 'same same but different' ) //DO I PUT SPACES OR NO?
+  } else {
+    console.log('different')
+  }
+}
 
-// console.log(sameSameButDifferent(5, 5)) 
-// // => "same same"
+console.log(sameSameButDifferent(5, 5)) 
+// => "same same"
 
-// console.log(sameSameButDifferent(123, "123")) 
-// // => "same same (but different)"
+console.log(sameSameButDifferent(123, "123")) 
+// => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 7)) 
-// // => "different"
+console.log(sameSameButDifferent(5, 7)) 
+// => "different"
 
-// console.log(sameSameButDifferent(123, "122")) 
-// // => "different"
-// console.log("%c----------", "color: red") 
+console.log(sameSameButDifferent(123, "122")) 
+// => "different"
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
+console.log("%cQuestion 5", "color: red")
+
+function updateGrade(student, grade) {
+  student['grade'] = grade   // WHAT EXACTLY IS THE CORRECT JSON OBJECT NOTATION. I TRIED [grade:] and [:grade] and they did not work
+}
+const student1 = { name: "Duane", grade: 88 }
+updateGrade(student1, 92)
+console.log(student1)  
 // // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red")
+console.log("%c----------", "color: red")
 
 
 
 // ***** Question 6 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
-//   }
-// ]
-// printNameAndPhones(users)
+console.log("%cQuestion 6", "color: red")
+
+
+function printNameAndPhones(users) {
+  users.forEach(function (e) {
+    console.log(e.name);
+    console.log(`Cell: ${e.phones['cell']}`);
+    console.log(`Office: ${e.phones['office']}`);
+  }
+  )
+}
+
```

Comment:
Looks good! I would recommend using a more semantic name for the anonymous function's parameter on line 98 (like `user`) 

- They're equally good options, but here's how it would look with for..of:
```javascript
 for (let user of users) {
  console.log(user.name);
  console.log(`Cell: ${user.phones.cell}`);
  console.log(`Office: ${user.phones.office}`);
 }
```